packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -27,388 +25,415 @@ export function reset() {
       }
+    },
+    [removeBodyTouchListeners],
+  );
 
-      return;
-    }
-
-    if (
-      this.velocity < -this.props.minFlingVelocity ||
-      1 - translateRatio > this.props.hysteresis
-    ) {
-      this.props.onOpen();
-    } else {
-      // Reset the position, the swipe was aborted.
-      this.setPosition(this.getMaxTranslate(), {
-        mode: 'enter',
-      });
+  React.useEffect(() => {
+    if (!open) {
+      setMaybeSwiping(false);
     }
-  };
+  }, [open]);
 
-  handleBackdropRef = ref => {
+  const handleBackdropRef = React.useCallback(instance => {
     // #StrictMode ready
-    this.backdropRef = ReactDOM.findDOMNode(ref);
-  };
+    backdropRef.current = ReactDOM.findDOMNode(instance);
+  }, []);
 
-  handlePaperRef = ref => {
+  const handlePaperRef = React.useCallback(instance => {
     // #StrictMode ready
-    this.paperRef = ReactDOM.findDOMNode(ref);
-  };
-
-  listenTouchStart() {
-    document.body.addEventListener('touchstart', this.handleBodyTouchStart);
-  }
-
-  removeTouchStart() {
-    document.body.removeEventListener('touchstart', this.handleBodyTouchStart);
-  }
-
-  removeBodyTouchListeners() {
-    document.body.removeEventListener('touchmove', this.handleBodyTouchMove, { passive: false });
-    document.body.removeEventListener('touchend', this.handleBodyTouchEnd);
-    document.body.removeEventListener('touchcancel', this.handleBodyTouchEnd);
-  }
-
-  render() {
-    const {
-      anchor,
-      disableBackdropTransition,
-      disableDiscovery,
-      disableSwipeToOpen,
-      hysteresis,
-      innerRef,
-      minFlingVelocity,
-      ModalProps: { BackdropProps, ...ModalPropsProp } = {},
-      onOpen,
-      open,
-      PaperProps = {},
-      SwipeAreaProps,
-      swipeAreaWidth,
-      variant,
-      ...other
-    } = this.props;
-    const { maybeSwiping } = this.state;
-
-    return (
-      <React.Fragment>
-        <Drawer
-          open={variant === 'temporary' && maybeSwiping ? true : open}
-          variant={variant}
-          ModalProps={{
-            BackdropProps: {
-              ...BackdropProps,
-              ref: this.handleBackdropRef,
-            },
-            ...ModalPropsProp,
-          }}
-          PaperProps={{
-            ...PaperProps,
-            style: {
-              pointerEvents: variant === 'temporary' && !open ? 'none' : '',
-              ...PaperProps.style,
-            },
-            ref: this.handlePaperRef,
-          }}
-          anchor={anchor}
-          ref={innerRef}
-          {...other}
-        />
-        {!disableSwipeToOpen && variant === 'temporary' && (
-          <NoSsr>
-            <SwipeArea
-              anchor={anchor}
-              innerRef={this.swipeAreaRef}
-              width={swipeAreaWidth}
-              {...SwipeAreaProps}
-            />
-          </NoSsr>
-        )}
-      </React.Fragment>
-    );
-  }
-}
+    paperRef.current = ReactDOM.findDOMNode(instance);
+  }, []);
```

Comment:
Would useCallback implicitly solve https://github.com/mui-org/material-ui/issues/10863 ?

Reply:
Can we try the issue reproduction with or without the defensive logic? We would know this way.

Reply:
Looks like we still need it